From 0e582aea5e4c1b4d34c93a23f00f5eac420ef585 Mon Sep 17 00:00:00 2001
From: Lewis Liu <wei.zhi.liu@intel.com>
Date: Tue, 11 Mar 2025 16:11:42 +0800
Subject: [PATCH 06/14] Fix launch error (#7)

* remove redundant dataset files.

Signed-off-by: Liu, Wei Zhi <wei.zhi.liu@intel.com>

* enable displaying for Examples and fix segment fault for stereo examples.

Signed-off-by: Liu, Wei Zhi <wei.zhi.liu@intel.com>

---------

Signed-off-by: Liu, Wei Zhi <wei.zhi.liu@intel.com>
---
 Examples/Monocular/mono_euroc.cc                  |  2 +-
 Examples/Stereo-Inertial/stereo_inertial_euroc.cc |  2 +-
 debian/rules                                      |  5 ++++-
 src/Settings.cc                                   | 10 ++++++----
 4 files changed, 12 insertions(+), 7 deletions(-)

diff --git a/Examples/Monocular/mono_euroc.cc b/Examples/Monocular/mono_euroc.cc
index 3a23312..1ea8f95 100644
--- a/Examples/Monocular/mono_euroc.cc
+++ b/Examples/Monocular/mono_euroc.cc
@@ -80,7 +80,7 @@ int main(int argc, char **argv)
     int fps = 20;
     float dT = 1.f/fps;
     // Create SLAM system. It initializes all system threads and gets ready to process frames.
-    ORB_SLAM3::System SLAM(argv[1],argv[2],ORB_SLAM3::System::MONOCULAR, false);
+    ORB_SLAM3::System SLAM(argv[1],argv[2],ORB_SLAM3::System::MONOCULAR, true);
     float imageScale = SLAM.GetImageScale();
 
     double t_resize = 0.f;
diff --git a/Examples/Stereo-Inertial/stereo_inertial_euroc.cc b/Examples/Stereo-Inertial/stereo_inertial_euroc.cc
index bb52223..d1ab311 100644
--- a/Examples/Stereo-Inertial/stereo_inertial_euroc.cc
+++ b/Examples/Stereo-Inertial/stereo_inertial_euroc.cc
@@ -129,7 +129,7 @@ int main(int argc, char **argv)
     cout.precision(17);
 
     // Create SLAM system. It initializes all system threads and gets ready to process frames.
-    ORB_SLAM3::System SLAM(argv[1],argv[2],ORB_SLAM3::System::IMU_STEREO, false);
+    ORB_SLAM3::System SLAM(argv[1],argv[2],ORB_SLAM3::System::IMU_STEREO, true);
 
     cv::Mat imLeft, imRight;
     for (seq = 0; seq<num_seq; seq++)
diff --git a/debian/rules b/debian/rules
index 5a1e417..f75ae92 100755
--- a/debian/rules
+++ b/debian/rules
@@ -33,7 +33,10 @@ override_dh_install:
 	cp Vocabulary/ORBvoc.txt debian/orb-slam3/opt/intel/orb-slam3/Vocabulary
 	cp -r Examples/* debian/orb-slam3/opt/intel/orb-slam3/Examples
 	find debian/orb-slam3/opt/intel/orb-slam3 -name "*.cc" | xargs rm
-
+	find debian/orb-slam3/opt/intel/orb-slam3/Examples/ -name "TUM_TimeStamps" | xargs rm -rf
+	find debian/orb-slam3/opt/intel/orb-slam3/Examples/ -name "TUM_IMU" | xargs rm -rf
+	rm -r debian/orb-slam3/opt/intel/orb-slam3/Examples/Monocular-Inertial/EuRoC_TimeStamps
+	rm -r debian/orb-slam3/opt/intel/orb-slam3/Examples/Monocular-Inertial/EuRoC_IMU
 
 # Clean up any temporary files that were generated during the build
 override_dh_clean:
diff --git a/src/Settings.cc b/src/Settings.cc
index 93a50be..4733d6e 100644
--- a/src/Settings.cc
+++ b/src/Settings.cc
@@ -558,11 +558,13 @@ namespace ORB_SLAM3 {
             else{
                 output << "Kannala-Brandt";
             }
-            output << "" << ": [";
-            for(size_t i = 0; i < settings.originalCalib2_->size(); i++){
-                output << " " << settings.originalCalib2_->getParameter(i);
+           if (settings.cameraType_ != Settings::Rectified){
+                output << "" << ": [";
+                for(size_t i = 0; i < settings.originalCalib2_->size(); i++){
+                    output << " " << settings.originalCalib2_->getParameter(i);
+                }
+                output << " ]" << endl;
             }
-            output << " ]" << endl;
 
             if(!settings.vPinHoleDistorsion2_.empty()){
                 output << "\t-Camera 1 distortion parameters: [ ";
-- 
2.30.2

